### ISSUE TITLE
Row ordering with sorting

### ISSUE THREAD

**Issue by ktriek** [KevinVandy/material-react-table#1241]
### material-react-table version

2 and 3 at least

### react & react-dom versions

17

### Describe the bug and the steps to reproduce it

The `draggingRow.index` and  `hoveredRow.index` are wrong when the table was previoulsy sorted on the first drop.
Future drops work as expected.

### Minimal, Reproducible Example - (Optional, but Recommended)

https://stackblitz.com/edit/github-r7odah?file=src%2FTS.tsx

### Screenshots or Videos (Optional)

![image](https://github.com/user-attachments/assets/dac858f1-d6e1-4826-be8a-c8b2c233ef77)

### Do you intend to try to help solve this bug with your own PR?

None

### Terms

- [X] I understand that if my bug cannot be reliably reproduced in a debuggable environment, it will probably not be fixed and this issue may even be closed.

**Comment by KevinVandy** [comment]
I usually assume that the sorting features should be turned off in MRT if you are enabling row ordering

**Comment by ktriek** [comment]
Yeah, that's what I understood from the storybooks, but we have this requirement, the users can sort by columns headers or customize the order(DnD).
